patterns/singleton.py

```diff
@@ -0,0 +1,17 @@
+"""
+Singleton Pattern Example
+Ensures a class has only one instance and provides a global point of access to it.
+"""
+
+class Singleton:
+    _instance = None
+
+    def __new__(cls):
+        if not cls._instance:
+            cls._instance = super(Singleton, cls).__new__(cls)
+        return cls._instance
+
+if __name__ == "__main__":
+    obj1 = Singleton()
```

Comment:
Also referring to the facade documentation lines 94-97, this repo uses doctests to establish testing. 

Could you follow the same pattern to demonstrate the principle you are demonstrating here..